### PR TITLE
[News] 2021-07-13

### PR DESCRIPTION
**Pull request by suetsuguk**
https://github.com/sider/runners/blob/0.52.0/CHANGELOG.md

### CODE REVIEW

**Review by ybiquitous — COMMENTED** [review]
Please add a new option `suppressions-list`.

**Comment by ybiquitous** [comment]
I think it better to include the changes of #632 and #637 into this PR.

File: docs/tools/cplusplus/cppcheck.md [hunk 1]
```diff
@@ -182,3 +190,14 @@ And, the results of the following checks are affected when the `parallel` option
...
+### `suppressions-list`
+
+This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:
+
+```text
+memleak
+uninitvar
+```
+
+You can read more about this file format in the [Cppcheck manual](https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file).
```
Comment (suetsuguk) [inline docs/tools/cplusplus/cppcheck.md:203]: > __[remark-lint]__ Link to https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file is dead ([view](https://sider.review/gh/repos/142577083/pulls/636#issue-5749131))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `no-dead-urls` | `warn` |
>
> You can [close this issue](https://sider.review/gh/repos/142577083/pulls/636#issue-5749131) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

The link is valid.

**Comment by suetsuguk** [comment]
> Please add a new option `suppressions-list`.

Added a note for `suppression-list`. commit e137daa175b1e2e798a0006f884c6f912f8a8da6

File: docs/news/2021.md [hunk 2]
```diff
@@ -11,6 +11,21 @@ This document describes notable changes on Sider in the year 2021.
...
 ---
 
+## Provide new recommended rulesets for Cppcheck and PHPMD
+
+<time class="news-date" datetime="2021-07-13">July 13, 2021</time>
+
+Today, we have just released our new recommended rulesets for [Cppcheck](../tools/cplusplus/cppcheck.md) and [PHPMD](../tools/php/phpmd.md).
+Please check out [Recommended Ruleset](../getting-started/recommended-rules.md) for more details about the rulesets.
+
+In addition, this update includes the following notable change: 💥
```
Comment (ybiquitous) [inline docs/news/2021.md:21]: [suggestion] I think adding an option is not notable and 💥 is usually used for breaking changes. How about this?

```suggestion
In addition, this release includes the following updates: ✨
```

File: docs/news/2021.md [hunk 3]
```diff
@@ -11,6 +11,21 @@ This document describes notable changes on Sider in the year 2021.
...
+
+- Support [`suppressions-list`](../tools/cplusplus/cppcheck.md#suppressions-list) option for Cppcheck - This option allows you to specify a file path of a suppressions file. This helps you filter out certain rules from being generated.
```
Comment (ybiquitous) [inline docs/news/2021.md:23]: [suggestion] I think this addition is not so important. So how about simplifying the description as below?

```suggestion
- Support [`suppressions-list`](../tools/cplusplus/cppcheck.md#suppressions-list) option for Cppcheck.
```

File: docs/tools/cplusplus/cppcheck.md [hunk 4]
```diff
@@ -40,23 +46,25 @@ linter:
     addon: "cert"
     bug-hunting: true
     parallel: true
+    suppressions-list: "suppressions.txt"
```
Comment (ybiquitous) [inline docs/tools/cplusplus/cppcheck.md:49]: [trivial]

```suggestion
    suppressions-list: "your_suppressions.txt"
```

File: docs/tools/cplusplus/cppcheck.md [hunk 5]
```diff
@@ -40,23 +46,25 @@ linter:
...
+| [`std`](#std)                                                                                 | `string`             | -                                |
+| [`project`](#project)                                                                         | `string`             | -                                |
+| [`language`](#language)                                                                       | `string`             | -                                |
+| [`addon`](#addon)                                                                             | `string`, `string[]` | -                                |
+| [`bug-hunting`](#bug-hunting)                                                                 | `boolean`            | `false`                          |
+| [`parallel`](#parallel)                                                                       | `boolean`            | `false`                          |
+| [`suppressions-list`](#suppressions-list)                                                     | `string`             | `sider_recommended_cppcheck.txt` |
```
Comment (ybiquitous) [inline docs/tools/cplusplus/cppcheck.md:67]: [suggestion] This default value is inaccurate. How about this?

```suggestion
| [`suppressions-list`](#suppressions-list)                                                     | `string`             | _(see below)_ |
```

File: docs/tools/cplusplus/cppcheck.md (hunk 1 above)
Comment (ybiquitous) [reply docs/tools/cplusplus/cppcheck.md:203]: [suggestion] https://github.com/sider/sider-docs/pull/636/files#r668400505

How about adding a link to `sider_recommended_cppcheck.txt` in this section?

**Comment by suetsuguk** [comment]
Thank you for review. Fixed in a single commit 3f6a1fe8b986acdafc11a67c3fbabe16a8dbddb2.

File: docs/tools/cplusplus/cppcheck.md (hunk 5 above)
Comment (suetsuguk) [reply docs/tools/cplusplus/cppcheck.md]: I also added this mark in the `rule` row of the PHPMD table.

File: docs/tools/cplusplus/cppcheck.md [hunk 6]
```diff
@@ -182,3 +190,16 @@ And, the results of the following checks are affected when the `parallel` option
...
+### `suppressions-list`
+
+This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:
+
+```text
+memleak
+uninitvar
+```
+
+If omitted, Sider uses the [recommended ruleset](https://github.com/sider/runners/blob/HEAD/images/cppcheck/sider_recommended_cppcheck.txt).
```
Comment (ybiquitous) [inline docs/tools/cplusplus/cppcheck.md:203]: [suggestion] How about avoiding duplication of the link to `sider_recommended_cppcheck.txt`?

```suggestion
If omitted, Sider uses the [recommended ruleset](#default-configuration-for-cppcheck).
```

File: docs/tools/php/phpmd.md [hunk 7]
```diff
@@ -92,7 +98,7 @@ Here are the available rulesets:
 - `naming`
 - `unusedcode`
 
-If omitted, Sider uses the [default configuration](https://github.com/sider/runners/blob/HEAD/images/phpmd/sider_config.xml).
+If omitted, Sider uses the [recommended ruleset](https://github.com/sider/runners/blob/HEAD/images/phpmd/sider_recommended_phpmd.xml).
```
Comment (ybiquitous) [inline docs/tools/php/phpmd.md:101]: [suggestion] Same here.

```suggestion
If omitted, Sider uses the [recommended ruleset](#default-configuration-for-phpmd).
```

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**